### DISCUSSION TITLE
Extract react-query dev tools as a browser extension

### DISCUSSION THREAD

**Discussion by DamianOsipiuk** [TanStack/query#1834] [Ideas] [answered]
Hi,

I'm currently trying to write some [Vue3 bindings](https://github.com/DamianOsipiuk/vue-react-query) for react-query.

This actually made me think about dev tools.
Currently, they are written using react and rely on the react context to work properly.
What I was thinking is to extract them from the core package and publish them as a browser extension. Similarly how redux dev tools work.

This will allow dev tools implementation to be used as a framework-agnostic addon.
Since dev tools actually need only core implementation to show what is happening under the hood, it will save a lot of hassle for other framework bindings to rewrite them per specific implementation.

What do you think?

**Comment by tannerlinsley** [comment, marked as answer]
There's a TON of work in building any devtools as a browser extensions vs inline like we have them today. I can't say I would wish that task on anyone right now simply because I tried to build it that way in the beginning and it was a nightmare. There is a reason they ended up the way the are today ;)

I think it would be better to first try and port the devtools to the desired framework and see what the similarities and difficulties are. Baby step.s

**Reply by DamianOsipiuk** [reply]
Seems reasonable ;) Thanks
